File: symfony/src/Service/Content/ArticleTagService.php
<?php

namespace App\Service\Content;

use App\Entity\Content\ArticleTag;
use App\Repository\Content\ArticleTagRepository;

class ArticleTagService
{
    public function __construct(
        private ArticleTagRepository $repository,
    ) {}

    public function create(array $data): ArticleTag
    {
        throw new \LogicException('Not implemented');
    }

    public function update(ArticleTag $entity, array $data): ArticleTag
    {
        throw new \LogicException('Not implemented');
    }

    public function rename(int $id, $newName): void
    {
        $entity = $this->repository->find($id);
        if (!$entity) throw new \RuntimeException('ArticleTag not found: ' . $id);
        $entity->rename($newName);
        $this->repository->save($entity, flush: true);
    }

    public function articleCount(int $id): mixed
    {
        $entity = $this->repository->find($id);
        if (!$entity) throw new \RuntimeException('ArticleTag not found: ' . $id);
        $result = $entity->articleCount();
        $this->repository->save($entity, flush: true);
        return $result;
    }
}

File: rails/app/services/content/article_tag_service.rb
module Content
  class ArticleTagService
    def initialize(repository = ArticleTag)
      @repository = repository
    end

    def create(attributes)
      raise NotImplementedError, "#{self.class}#create not implemented"
    end

    def update(entity, attributes)
      raise NotImplementedError, "#{self.class}#update not implemented"
    end

    def rename(id, new_name)
      instance = ArticleTag.find(id)
      instance.rename(new_name)
      instance.save!
    end

    def article_count(id)
      instance = ArticleTag.find(id)
      result = instance.article_count()
      instance.save!
      result
    end
  end
end
